File: 225. Implement Stack using Queues/Solution.java
class MyStack {
    Queue<Integer> q1 = new ArrayDeque<>();
    Queue<Integer> q2 = new ArrayDeque<>();
    public MyStack() {
        
    }
    
    public void push(int x) {
        q1.add(x);
    }
    
    public int pop() {
        while(q1.size()>1){
            q2.add(q1.peek());
            q1.remove();
        }
        int a = q1.peek();
        q1.remove();
        while(q2.size()!=0){
            q1.add(q2.peek());
            q2.remove();
        }
        return a;
    }
    
    public int top() {
        while(q1.size()>1){
            q2.add(q1.peek());
            q1.remove();
        }
        int a = q1.peek();
        q2.add(q1.peek());
        q1.remove();
        while(q2.size()>0){
            q1.add(q2.peek());
            q2.remove();
        }
        return a;
    }
    
    public boolean empty() {
        if(q1.size()==0) return true;
        return false;
    }
}

/**
 * Your MyStack object will be instantiated and called as such:
 * MyStack obj = new MyStack();
 * obj.push(x);
 * int param_2 = obj.pop();
 * int param_3 = obj.top();
 * boolean param_4 = obj.empty();
 */

File: 225. Implement Stack using Queues/Solution.py
class MyStack:

    def __init__(self):
        self.queue_1 = []
        self.queue_2 = []

    def push(self, x: int) -> None:
        self.queue_1.append(x)

    def pop(self) -> int:
        while len(self.queue_1)>1:
            self.queue_2.append(self.queue_1.pop(0))
        ret = self.queue_1.pop()
        while len(self.queue_2)>0:
            self.queue_1.append(self.queue_2.pop(0))
        return ret

    def top(self) -> int:
        while len(self.queue_1)>1:
            self.queue_2.append(self.queue_1.pop(0))
        ret = self.queue_1.pop()
        while len(self.queue_2)>0:
            self.queue_1.append(self.queue_2.pop(0))
        self.queue_1.append(ret)
        return ret

    def empty(self) -> bool:
        return len(self.queue_1)==0
    
# Your MyStack object will be instantiated and called as such:
# obj = MyStack()
# obj.push(x)
# param_2 = obj.pop()
# param_3 = obj.top()
# param_4 = obj.empty()